CHANGELOG.md

```diff
@@ -1,3 +1,24 @@
+## 0.12.0 (April 22, 2024)
+
+NOTES:
+* **This release requires juju controller version 3.4 or later juju.**
+* This release uses juju client api code from the juju 3.5-beta1 candidate release.
+
+ENHANCEMENTS:
+* Schema and methods for user secrets management are implemented.
+* `resource_secret` and `resource_access_secret` resources are added.
+* Add grant and remove secret access function to `internal/juju`.
+* Data source for secrets is added.
+* Client ID and secret are added to the provider config.
+* Update, and remove run limitations for build and unit test workflows.
+* README.md and GitHub issue templates are updated.
+* github.com/juju/names/v4 is updated to v5.
+
+BUG FIXES:
+
+* Validate channel in application resource by @Aflynn50 in https://github.com/juju/terraform-provider-juju/pull/447
```

Comment:
We should call this out more - due to the change in channel for charms requiring both track and risk now.